src/main/java/com/aws/greengrass/clientdevices/auth/configuration/PolicyVariableResolver.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.configuration;
+
+import com.aws.greengrass.clientdevices.auth.exception.PolicyException;
+import com.aws.greengrass.clientdevices.auth.session.Session;
+import com.aws.greengrass.util.Pair;
+import org.apache.commons.lang3.StringUtils;
+import software.amazon.awssdk.utils.ImmutableMap;
+
+import java.util.List;
+import java.util.Map;
+
+public final class PolicyVariableResolver {
+    private static final String THING_NAMESPACE = "Thing";
+    private static final String THING_NAME_ATTRIBUTE = "ThingName";
+
+    private static final Map<String, Pair<String,String>> policyVariableToAttributeProvider = ImmutableMap.of(
+            "${iot:Connection.Thing.ThingName}", new Pair<>(THING_NAMESPACE, THING_NAME_ATTRIBUTE)
+    );
+
+    private PolicyVariableResolver() {
+    }
+
+    /**
+     * Utility method to replace policy variables in permissions with device attributes.
+     * Policy variables need to be validated when reading the policy document.
+     * This method does not handle unsupported policy variables.
+     *
+     * @param policyVariables list of policy variables in permission format
+     * @param format permission format to resolve
+     * @param session current device session
+     * @return updated format
+     * @throws PolicyException when unable to find a policy variable value
+     */
+    public static String resolvePolicyVariables(List<String> policyVariables, String format, Session session)
```

Comment:
you mentioned before, but we can make this a set now

Reply:
done